### PR TITLE
support aquct and co2store

### PR DESCRIPTION
**Pull request by totto82**
Depends on OPM/opm-material#471 and OPM/opm-common#2694

If you want to run the test case in opm-tests you will also need OPM/opm-common#2691

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
Any comments on this? I plan to merge this later today if no objections.

File: opm/simulators/aquifers/AquiferCarterTracy.hpp [hunk 1]
```diff
@@ -191,7 +215,20 @@ class AquiferCarterTracy : public AquiferInterface<TypeTag>
...
         this->pa0_ = this->aquct_data_.initial_pressure.value();
-        this->rhow_ = this->aquct_data_.waterDensity();
+        if(this->co2store_()) {
+             const auto press = this->aquct_data_.initial_pressure.value();
+
+             Scalar temp = FluidSystem::reservoirTemperature();
+             if (this->aquct_data_.initial_temperature.has_value())
+                 temp = this->aquct_data_.initial_temperature.value();
+
+             Scalar rs = 0.0; // no dissolved CO2
```
Comment (bska) [inline opm/simulators/aquifers/AquiferCarterTracy.hpp:225]: The `rs` variable is unused.  Did you mean to pass this as an argument to `inverseFormationVolumeFactor()`?

**Review by bska — COMMENTED** [review]
Why do we need to use the `oilPvt()` to calculate water properties in the CO2STORE case?  The aquifer is still primarily filled with water is it not?

**Comment by totto82** [comment]
> Why do we need to use the oilPvt() to calculate water properties in the CO2STORE case? The aquifer is still primarily filled with water is it not?

Currently only the oil phase supports dissolved gas therefore CO2STORE "use" OIL for the water phase. This is the standard trick used for doing co2 storage simulations using a blackoil simulator.

File: opm/simulators/aquifers/AquiferCarterTracy.hpp (hunk 1 above)
Comment (totto82) [reply opm/simulators/aquifers/AquiferCarterTracy.hpp:225]: yes. thanks for noticing. I will update accordingly.

File: opm/simulators/aquifers/AquiferCarterTracy.hpp [hunk 2]
```diff
@@ -159,6 +166,11 @@ class AquiferCarterTracy : public AquiferInterface<TypeTag>
         return std::make_pair(a, b);
     }
 
+    size_t pvtRegionIdx() const
```
Comment (bska) [inline opm/simulators/aquifers/AquiferCarterTracy.hpp:169]: Please use `std::size_t` here instead.

**Comment by totto82** [comment]
jenkins build this please

**Pull request merged**